### PR TITLE
TEST-#3474: fix pip install commmand in push-to-master

### LINKED ISSUE #3474: New pip resolver takes a very long time to complete
Issue on pip side https://github.com/pypa/pip/issues/9187.

Problem in our CI (takes more that 5 hours) https://github.com/modin-project/modin/runs/3695414357.

Good comment about possible solutions https://github.com/pypa/pip/issues/9187#issuecomment-736000492.

I propose the easiest way to support - to use the old resolver, until the pip optimizes the new resolver.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [x] Resolves #3474 
- [ ] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: .github/workflows/push-to-master.yml [hunk 1]
```diff
@@ -1,79 +1,6 @@
 name: push-to-master
-on:
-  push:
-    branches:
-      - master
+on: pull_request
 jobs:
-  test-ray-master:
-    runs-on: ubuntu-latest
```
Comment (vnlitvinov) [inline .github/workflows/push-to-master.yml:8]: why change this?

Comment (anmyachev) [reply .github/workflows/push-to-master.yml:8]: For test purpose; this should be reverted before merge.

File: .github/workflows/push-to-master.yml [hunk 2]
```diff
@@ -115,7 +42,7 @@ jobs:
           python-version: ${{matrix.python-version}}
           channel-priority: strict
           use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
-      - run: pip install -r requirements-dev.txt
+      - run: pip install -r requirements-dev.txt --use-deprecated=legacy-resolver
```
Comment (anmyachev) [inline .github/workflows/push-to-master.yml:45]: This will be the only change.

Comment (vnlitvinov) [reply .github/workflows/push-to-master.yml:45]: okay, revert the unnecessary changes then

**Review by vnlitvinov — APPROVED** [review]
Not looking good, but would solve as a stopgap I think...

**Pull request merged**